src/main/groovy/com/twcable/gradle/sling/osgi/BundleAndServers.groovy

```diff
@@ -0,0 +1,553 @@
+/*
+ * Copyright 2014-2015 Time Warner Cable, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.twcable.gradle.sling.osgi
+
+import com.twcable.gradle.http.HttpResponse
+import com.twcable.gradle.http.SimpleHttpClient
+import com.twcable.gradle.sling.SimpleSlingSupportFactory
+import com.twcable.gradle.sling.SlingServerConfiguration
+import com.twcable.gradle.sling.SlingServersConfiguration
+import com.twcable.gradle.sling.SlingSupport
+import com.twcable.gradle.sling.SlingSupportFactory
+import groovy.json.JsonSlurper
+import groovy.transform.TypeChecked
+import groovy.util.logging.Slf4j
+import org.apache.http.entity.mime.content.FileBody
+import org.gradle.api.GradleException
+
+import javax.annotation.Nonnull
+import javax.annotation.Nullable
+
+import static BundleState.ACTIVE
+import static BundleState.FRAGMENT
+import static BundleState.INSTALLED
+import static BundleState.MISSING
+import static BundleState.RESOLVED
+import static SlingSupport.block
+import static java.net.HttpURLConnection.HTTP_BAD_METHOD
+import static java.net.HttpURLConnection.HTTP_BAD_REQUEST
+import static java.net.HttpURLConnection.HTTP_CLIENT_TIMEOUT
+import static java.net.HttpURLConnection.HTTP_OK
+
+/**
+ * Brings together a bundle configuration and a set of Sling servers to provides for interaction via the Sling REST API.
+ */
+@Slf4j
+@TypeChecked
+@SuppressWarnings(["GroovyPointlessBoolean", "GrMethodMayBeStatic"])
+class BundleAndServers {
```

Comment:
I just left everything `@TypedChecked` since it's running in the context of Gradle and its dynamic calls. Makes no real difference either way, as long as there's some sort of static typing.